include/bout/invert/laplacexz.hxx

```diff
@@ -38,7 +38,8 @@
 
 class LaplaceXZ {
 public:
-  LaplaceXZ(Mesh *UNUSED(m), Options *UNUSED(options), const CELL_LOC UNUSED(loc)) {}
+  LaplaceXZ(Mesh* UNUSED(m), Options* UNUSED(options), const CELL_LOC loc = CELL_CENTRE)
```

Comment:
Don't think we need the default argument here, because the constructor(s) should only ever be called through `LaplaceXZ::create`, which already has the default argument set.

Reply:
We'd like users to use `create` I guess, but there's nothing currently stopping a user from using the exposed constructor.

Reply:
I suspect this code may need a refactor in the long run to look a bit more like our usual factory methods.

Reply:
OK, that's a good point, and having default arguments in the constructors is also what we do for Laplacian and it's implementations. We need default arguments for the implementations' constructors then, I'll add those while I'm at it.

Agree that a refactor to be like the usual factory methods would be good :+1: